### PR TITLE
Create github-ci.yml

### PR DESCRIPTION
**Pull request by JFinneganAloxy**

### CODE REVIEW

File: .github/workflows/github-ci.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    - name: Handle results
+      run: |
+        if ! grep -q 'AES all unit tests OK !' "results_aes.txt"; then exit 1;  fi
+        if ! grep -q 'Unit-tests for ALP completed' "results_alp.txt"; then exit 1;  fi
+        if ! grep -q 'Input was decoded successfully' "results_fec.txt"; then exit 1;  fi
+        if ! grep -q 'All FIFO tests passed!' "results_fifo.txt"; then exit 1;  fi   
+
+  run-testsuite:
+    name: Run Testsuite
+    runs-on: self-hosted
```
Comment (LOorts-Aloxy) [inline .github/workflows/github-ci.yml:51]: is it actually possible to run different steps on different runners? for example: building the images could be done on a docker runner but flashing and testing has to run on self-hosted

**Review by LOorts-Aloxy — APPROVED** [review]
lgtm, nice work!

**Comment by JFinneganAloxy** [comment]
Still need to add some sort of check to ensure the devices are not in use.

File: .github/workflows/github-ci.yml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    needs: generate-builds
+
+    steps:
+    - uses: actions/download-artifact@v3
+
+    - name: Flash builds to devices
+      run: |
+        if [ lsof -i -P -n | grep -q JLinkExe ] ; then exit 1; fi
+        if [ lsof -n /dev/ttyUSB0 | grep -q /dev/ttyUSB0 ] ; then exit 1; fi
+        if [ lsof -n /dev/ttyUSB1 | grep -q /dev/ttyUSB1 ] ; then exit 1; fi
```
Comment (LOorts-Aloxy) [inline .github/workflows/github-ci.yml:90]: also do this test when running the actual testsuite, could be the tests are triggered at the exact same time

**Pull request merged**